lib/active_model/serializer/adapter/json.rb

```diff
@@ -4,44 +4,122 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
 
         def serializable_hash(options = nil)
           options ||= {}
+          serialized_hash = {}
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            serialized_hash = serialize_array_without_root(serializer, options)
           else
-            hash = {}
+            serialized_hash = resource_object_for(serializer)
+            serialized_hash = add_resource_relationships(serialized_hash, serializer)
```

Comment:
I agree with you. What would be best, I think, would be to rename it to `resource_relationships_for(serializer)` and merge the resulting hashes. Would that be possible? (I don't know much about the JSON adapter, so it might very well not be the case).